x-pack/plugins/ml/public/application/model_management/model_actions.tsx

```diff
@@ -431,14 +446,15 @@ export function useModelActions({
         color: 'danger',
         isPrimary: false,
         onClick: (model) => {
-          onModelsDeleteRequest([model.model_id]);
+          onModelsDeleteRequest([model]);
         },
         available: (item) =>
-          canDeleteTrainedModels && !isBuiltInModel(item) && !item.putModelConfig,
+          canDeleteTrainedModels &&
+          !isBuiltInModel(item) &&
+          !item.putModelConfig &&
+          (Object.keys(item.pipelines ?? {}).length === 0 || canManageIngestPipelines),
```

Comment:
nit, it might be easier to read if this was moved to a function.

Reply:
updated in https://github.com/elastic/kibana/pull/158671/commits/7079f7f24486f7657b316b776c95a80af3b45bf0